homeassistant/components/imap/coordinator.py

```diff
@@ -492,11 +494,11 @@ async def _async_wait_push_loop(self) -> None:
                 self.auth_errors = 0
                 self.async_set_updated_data(self.number_of_messages)
             try:
-                idle: asyncio.Future = await self.imap_client.idle_start()
+                self._idle = await self.imap_client.idle_start()
                 await self.imap_client.wait_server_push()
                 self.imap_client.idle_done()
                 async with asyncio.timeout(10):
-                    await idle
+                    await self._idle
```

Comment:
It seems `Abort` is raised, a subclass or `AioImapException`. To avoid the future to be garbadge collected too soon, we could make tie it to the class

Reply:
I don't think its possible for the future to be garage collected while we are awaiting it